### ISSUE TITLE
HttpClientInterface withOptions methods make error in dev

### ISSUE THREAD

**Issue by Kleinast** [symfony/symfony#46036]
### Symfony version(s) affected

4.4.39

### Description

I implement a custom client using the httpClientInterface
```
final class ApiClient
{
    const API_URL = 'https://***.com/v1';

    private HttpClientInterface $httpClient;

    public function __construct(
        string $clientId,
        string $apiKey,
        HttpClientInterface $httpClient
    )
    {
        $this->httpClient = $httpClient->withOptions([
            'base_uri' => self::API_URL,
            'auth_basic' => "{$clientId}:{$apiKey}",
            'headers' => ['Content-Type' => 'application/json'],
        ]);
    }
```
And I have this error:
`Attempted to call an undefined method named "withOptions" of class "Symfony\Component\HttpClient\TraceableHttpClient".`

And indeed, when I look at the TraceableHttpClient, there is no withOptions method set, but for me as it is declared in HttpClientInterface, it should be implemented as well in TraceableHttpClient, isn't it ?

Sorry in advance if it's me doing a mistake

### How to reproduce

Call HttpClientInterface $httpClient->withOptions() methid in symfony dev mode where service autowire is Symfony\Component\HttpClient\TraceableHttpClient

**Comment by Kleinast** [comment]
For info, I just tryed to update symfony/http-client to 5.4 and the method has been implemented in this version so it work on 5.4.x

**Comment by xabbuh** [comment]
This method was added to the interface in version 3.0.0 of the HttpClient contracts. Likewise it was added to the `TraceableHttpClient` in 5.3.0. So yeah, if you want to depend on this method, you will need to update the HttpClient version that you use.
